common/client.go

```diff
@@ -10,9 +10,28 @@ import (
 	"github.com/databricks/databricks-sdk-go"
 	"github.com/databricks/databricks-sdk-go/client"
 	"github.com/databricks/databricks-sdk-go/config"
+	"github.com/databricks/databricks-sdk-go/service/scim"
 	"github.com/golang-jwt/jwt/v4"
 )
 
+type CachedCurrentUserService struct {
+	client     *client.DatabricksClient
+	oldImpl    scim.CurrentUserService
+	cachedUser *scim.User
+}
+
+func (a *CachedCurrentUserService) Me(ctx context.Context) (*scim.User, error) {
+	if a.cachedUser.DisplayName != "" {
+		return a.cachedUser, nil
+	}
+	user, err := a.oldImpl.Me(ctx)
+	if err != nil {
+		return user, err
+	}
+	*a.cachedUser = *user
```

Comment:
why not `a.cachedUser = user`?..

Reply:
The pointer points to the cached user in client. This is why we need to copy into the cachedUser value.

Reply:
just make it `cachedUser     *scim.User` on the client.

Reply:
That would not work I think. If we do this, then `a.cachedUser = user` will lead to the cachedUser within `CachedUserService` to point to the new user, but the changes won't be propagated to the client.

Reply:
i don't understand it